lib/hooks.js

```diff
@@ -101,32 +94,18 @@ function getHtmlWebpackPluginHooks (compilation) {
  * Add hooks to the webpack compilation object to allow foreign plugins to
  * extend the HtmlWebpackPlugin
  *
- * @param {WebpackCompilation} compilation
+ * @returns {HtmlWebpackPluginHooks}
  */
-function attachHooksToCompilation (compilation) {
-  /** @type {HtmlWebpackPluginHooks} */
-  const hooks = compilation.hooks;
-  hooks.htmlWebpackPluginBeforeHtmlGeneration = new AsyncSeriesWaterfallHook(['pluginArgs']);
-  hooks.htmlWebpackPluginBeforeHtmlProcessing = new AsyncSeriesWaterfallHook(['pluginArgs']);
-  hooks.htmlWebpackPluginAlterAssetTags = new AsyncSeriesWaterfallHook(['pluginArgs']);
-  hooks.htmlWebpackPluginAfterHtmlProcessing = new AsyncSeriesWaterfallHook(['pluginArgs']);
-  hooks.htmlWebpackPluginAfterEmit = new AsyncSeriesWaterfallHook(['pluginArgs']);
+function createHtmlWebpackPluginHooks () {
+  return {
+    htmlWebpackPluginBeforeHtmlGeneration: new AsyncSeriesWaterfallHook(['pluginArgs']),
```

Comment:
You should consider removing the `htmlWebpackPlugin` prefix, as it's now already namespace via `getHooks`.

Reply:
👍 I'll take a look